src/Features/CSharp/Portable/MoveToResx/CSharpMoveToResxCodeFixProvider.cs

```diff
@@ -0,0 +1,474 @@
+        => new CSharpMoveToResxFixAllProvider();
+
+    public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+    {
+        var diagnostic = context.Diagnostics[0];
+        var root = await context.Document.GetRequiredSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+        var node = root.FindNode(diagnostic.Location.SourceSpan, getInnermostNodeForTie: true) as LiteralExpressionSyntax;
+        if (node is null)
+            return;
+
+        var allResx = GetAllResxFiles(context.Document.Project).ToList();
+        if (allResx.Count == 0)
+        {
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    title: CSharpFeaturesResources.No_resx_file_found_Please_create_a_resx_file_in_your_project_or_folder,
+                    createChangedDocument: c => Task.FromResult(context.Document), // No-op
+                    equivalenceKey: "NoResxFileFound"),
+                diagnostic);
+            return;
+        }
+
+        var nestedActions = allResx
+            .Select(resx =>
+            {
+                var resxName = Path.GetFileName(resx.FilePath) ?? CSharpFeaturesResources.Unnamed_Resource;
+                return CodeAction.Create(
+                    title: resxName,
+                    createChangedSolution: c => MoveStringToResxAndReplaceLiteralAsync(context.Document, node, resx, c),
+                    equivalenceKey: $"MoveStringTo_{resxName}");
+            })
+            .ToImmutableArray();
+
+        context.RegisterCodeFix(
+            CodeAction.Create(
+                CSharpFeaturesResources.Move_string_to_resx_resource,
+                nestedActions,
+                isInlinable: false),
+            diagnostic);
+    }
+
+    private static IEnumerable<TextDocument> GetAllResxFiles(Project project)
+    {
+        return project.AdditionalDocuments
+            .Where(f => f.FilePath != null && string.Equals(".resx", Path.GetExtension(f.FilePath), StringComparison.OrdinalIgnoreCase));
+    }
+
+    private static async Task<Solution> MoveStringToResxAndReplaceLiteralAsync(
+        Document document,
+        LiteralExpressionSyntax stringLiteral,
+        TextDocument resxFile,
+        CancellationToken cancellationToken)
+    {
+        var value = stringLiteral.Token.ValueText;
+        var resourceKey = ToDeterministicResourceKey(value);
+
+        var resourceOperation = new ResourceOperation(value, resourceKey);
+        var replacementOperation = new ReplacementOperation(stringLiteral, resourceKey);
+
+        return await ResxResourceManager.UpdateResxAndDocumentAsync(
+            document,
+            resxFile,
+            [resourceOperation],
+            [replacementOperation],
+            cancellationToken).ConfigureAwait(false);
+    }
+
+    private static string ToDeterministicResourceKey(string value, int maxLength = 60)
+    {
+        if (string.IsNullOrWhiteSpace(value))
+            return "emptyString";
```

Comment:
we should nto be offering to update the empty string to a resx resource in the first place. that will just be noisy.